core/src/com/unciv/logic/GameStarter.kt

```diff
@@ -176,10 +176,18 @@ object GameStarter {
 
     private fun addCivStartingUnits(gameInfo: GameInfo) {
 
+        val ruleSet = gameInfo.ruleSet
+        val startingEra = gameInfo.gameParameters.startingEra
+        var startingUnits: MutableList<String>
+        var eraUnitReplacement: String
+        
         val startingLocations = getStartingLocations(
                 gameInfo.civilizations.filter { !it.isBarbarian() },
                 gameInfo.tileMap)
-        
+
+        val settlerLikeUnits = ruleSet.units.filter {
+            it.value.uniqueObjects.any { it.placeholderText == Constants.settlerUnique }
```

Comment:
ruleSet.units.values.filter{ it.hasUnique (Constants.settlerUnique) } perhaps? Don't recall if hasUnique is also on baseunit or only on mapunit

Reply:
baseUnit has no hasUnique or getMatchingUniques, sadly. I can add them if you'd like though

Reply:
No need